clients/rospy/src/rospy/core.py

```diff
@@ -215,6 +224,29 @@ def __call__(self, caller_id, period):
 _logging_throttle = LoggingThrottle()
 
 
+class LoggingUnique(object):
```

Comment:
There's a lot of DRY here with the existing throttle functionality. I would suggest passing one more 'throttle_identical' parameter to the _logging_throttle callable and amalgamate the functionality.

Reply:
Would also need the msg, which it doesn't currently receive despite it's internal doc. While the basic structure of each is similar they are not looking for the same values, so each dictionary would either need to be expanded or we'd still need two dictionaries. I don't think the logic loops will combine cleanly enough to merit merging.

Reply:
i could rewrite like so:
```
class LoggingThrottle(object):

    last_logging_time_table = {}
    last_logging_msg_table = {}

    def throttle_period(self, caller_id, period):
        """Do logging specified message periodically.

        - caller_id (str): Id to identify the caller
        - period (float): Period to do logging in second unit.
        """
        now = rospy.Time.now()

        last_logging_time = self.last_logging_time_table.get(caller_id)

        if (last_logging_time is None or
              (now - last_logging_time) > rospy.Duration(period)):
            self.last_logging_time_table[caller_id] = now
            return True
        return False
    
    def throttle_identical(self, caller_id, msg):
        """Do logging specified message only if distinct from last message.

        - caller_id (str): Id to identify the caller
        - msg (str): Contents of message
        """
        msg_hash = md5(msg).hexdigest()

        last_logging_hash = self.last_logging_msg_table.get(caller_id)

        if last_logging_hash is None or last_logging_hash != msg_hash:
            self.last_logging_msg_table[caller_id] = msg_hash
            return True
        return False


_logging_throttle = LoggingThrottle()
```

Reply:
then instead of referencing _logging_throttle or _logging_unique you reference _logging_throttle.throttle_period or _logging_throttle.throttle_identical

Reply:
At second view, it's fine as is, thanks.